docs/builder_author_faq.md

```diff
@@ -16,3 +16,56 @@ Use `build_to: cache` when:
 - The builder targets use cases for "application" packages that are unlikely to
   be published rather than packages which will be published and become
   dependencies.
+
+## How can I have temporary outputs only used during the Build?
+
+Due to build restrictions - namely that a builder can't read the outputs
+produced by other builders in the same phase - it's sometimes necessary to write
+information to a temporary file in one phase, then read that in a subsequent
+phase, but the intermediate result is not a useful output outside of the build.
+
+Use a `PostProcessBuilder` to "delete" files so they are not included in the
+merged output directory or available through the development server. Note that
+files are never deleted from disk, instead a "delete" by a `PostProcessBuilder`
+acts a filter on what assets can be see in the result of the build. This works
+best if temporary assets have a unique extension. The `FileDeletingBuilder` from
+the `build` package is designed for this case and only needs to be configured
+with the extensions it should remove. In some cases the builder should only
+operate in release mode so the files can see be seen in development mode - use
+the `isEnabled` argument to the constructor rather than returning a different
+builder or passing a different set of extensions - if the extensions change
+between modes it will invalidate the entire build.
+
+For example:
+
+```dart
+// In lib/builder.dart
+PostProcessBuilder temporaryFileCleanup(BuilderOptions options) =>
+    const FileDeletingBuilder(const ['.used_during_build'],
+        isEnabled: options.config['enabled'] as bool ?? false);
+Builder writesTemporary([_]) => ...
+Builder readsTemporaryWritesPermanent([_]) => ...
+```
+
+```yaml
+builders:
+  my_builder:
+    import: "package:my_package/builders.dart"
+    builder_factories:
+       - writesTemporary
+       - readsTemporaryWritesPermanent
+    build_extensions: {".dart": [".used_during_build", ".output_for_real"]}
```

Comment:
nit: remove one of these (keep the one that omits `{` and `[` imo)

Reply:
Ha, I though you'd like the second version better which is why I added it, then I forgot to go back and delete the `{}` 😆